src/edifact/parser.ts

```diff
@@ -0,0 +1,172 @@
+ * 
+ *  The implementation is based on the documentation from
+ *
+ *  https://www.gs1.org/docs/EDI/eancom/2012/ean02s3/part1/part1_05.htm
+ */
+export default function parse(str: string): Interchange {
+    // first line decides which separators are to be used in the rest of the interchange
+    const ssa = parseServiceStringAdvice(str)
+    const segments = split(str, ssa.segmentTerminator, ssa.releaseCharacter)
+
+    let header = null
+    const messages: Message[] = []
+
+    let messageHeader = null
+    let messageSegments: Segment[] = []
+
+    for (let i = 0; i < segments.length; ++i) { 
+        // line feeds between segments should be ignored according to the docs
+        const segment = segments[i].replace(/^(\r\n|\n|\r)|(\r\n|\n|\r)$/g,"")
+        // we already took care of the service string advice...
+        if (i == 0 && segment.startsWith("UNA")) continue
+
+        // array of elements which in turn each contain an array of components
+        const elements: Element[] = split(segment, ssa.elementSeparator, ssa.releaseCharacter).map(
+            (element) => split(element, ssa.componentSeparator, ssa.releaseCharacter).map(
+                (component) => unescape(component, ssa.releaseCharacter)
+            )
+        )
+
+        const segmentTag = elements.shift()!
+        if (segmentTag.length > 1) {
+            throw new Error("Nesting of segments is not supported by this parser")
+        }
+        const segmentCode = segmentTag[0]
+        
+        switch(segmentCode) {
+            // Interchange header
+            case "UNB": 
+                const syntaxIdentifier = elements[0][0]
+                const syntaxVersionNumber = elements[0][1]
+                if (syntaxIdentifier != "UNOC" || syntaxVersionNumber != "3") {
+                    throw new Error("Only syntax identifier UNOC version 3 is supported by this parser")
+                }
+                header = elements
+                break
+            // Interchange trailer
+            case "UNZ": 
+                break
+
+            // functional group header 
+            case "UNG": 
+            // functional group trailer
+            case "UNE": 
+                throw new Error("Functional groups (UNG, UNE) are not supported by this parser")
+
+            // message header
+            case "UNH": 
+                messageHeader = elements
+                break
+            // message trailer
+            case "UNT": 
+                messages.push({
+                    header: messageHeader!,
+                    segments: messageSegments!
+                })
+                messageHeader = null
+                messageSegments = []
+                break
+
+            // all other
+            default: 
+                messageSegments.push({
+                    tag: segmentCode,
+                    elements: elements
+                })
+                break
+        }
+    }
```

Comment:
In case we agree on using forEach…
```suggestion
    })
```